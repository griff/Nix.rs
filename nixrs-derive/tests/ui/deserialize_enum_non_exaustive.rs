use nixrs_derive::NixDeserialize;

#[derive(NixDeserialize)]
pub enum Test {
    #[nix(version = "..=10")]
    Old,
    #[nix(version = "15..=17")]
    Legacy,
    #[nix(version = "50..")]
    NewWay,
}

fn main() {}
