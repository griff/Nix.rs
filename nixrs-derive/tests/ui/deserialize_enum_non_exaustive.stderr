error[E0004]: non-exhaustive patterns: `11_u8..=14_u8` and `18_u8..=49_u8` not covered
 --> tests/ui/deserialize_enum_non_exaustive.rs:3:10
  |
3 | #[derive(NixDeserialize)]
  |          ^^^^^^^^^^^^^^ patterns `11_u8..=14_u8` and `18_u8..=49_u8` not covered
  |
  = note: the matched value is of type `u8`
  = note: this error originates in the derive macro `NixDeserialize` (in Nightly builds, run with -Z macro-backtrace for more info)
